src/functions.php

```diff
@@ -502,7 +502,8 @@ function watch(string|array $path, callable $function, ?Context $context = null)
     $context ??= GlobalHelper::getContext();
 
     $binary = match (true) {
-        OsHelper::isMacOS() => 'watcher-darwin',
+        OsHelper::isMacOS() && php_uname('m') === 'arm64' => 'watcher-darwin-arm64',
+        OsHelper::isMacOS() => 'watcher-darwin-amd64',
```

Comment:
This could also be done like this:
```suggestion
        OsHelper::isMacOS() => match(php_uname('m')) {
            'arm64' => 'watcher-darwin-arm64',
            default => 'watcher-darwin-amd64',
        },
```